src/vos/tests/vts_ilog.c

```diff
@@ -554,16 +547,19 @@ ilog_test_update(void **state)
 	rc = entries_check(umm, ilog, &ilog_callbacks, NULL, 0, entries);
 	assert_int_equal(rc, 0);
 
-	/** Same epoch, different DTX */
+	/** Same epoch, different DTX, same operation */
```

Comment:
Some confused, consider the subsequent test, how to distinguish "same operation" and "different operation"? Just "update" or "punch"?
If two different DTXs use the same epoch to "update" the same target, one of them will get "-DER_ALREADY", right? Why not get "-DER_TX_RESTART" to make such DTX to restart with newer epoch?

Reply:
The ALREADY will only be returned if the transaction isn't using dtx.  Otherwise, it will get DER_TX_RESTART